setup.py

```diff
@@ -435,8 +433,14 @@ def run(self):
 else:
     extra_compile_args = ["-Werror"]
     extra_link_args = []
-    if debugging_symbols_requested:
-        extra_compile_args.append("-g")
+    if not debugging_symbols_requested:
```

Comment:
I guess Python by default (at least locally and looking at some of the CI builds) includes the `-g` flag as part of the CFLAGS that distutils uses to compile extensions, so in the current setup this does nothing. 

According to SO we can override that by appending here, which might help reduce file size by removing those symbols:

https://stackoverflow.com/a/37952343/621736

I can also remove this from this PR if deemed too orthogonal. IIRC @xhochy or @TomAugspurger may have experience with stripping debug symbols from built distributions

Reply:
Multibuild may do this by default now? I don't recall.

Reply:
Yes, multibuild includes this nowadays.